src/sentry/utils/snuba.py

```diff
@@ -900,7 +930,11 @@ def constrain_column_to_dataset(col, dataset, value=None):
     unknown columns into tags expressions.
     """
     if col.startswith("tags["):
-        return col
+        if dataset == Dataset.Events:
+            return col
+        else:  # This makes sure already wrapped tags are pointed to the events table.
+            return u"events.{}".format(col)
```

Comment:
This branch shouldn't run for transactions dataset, or discover. Instead this looks like it is only applicable for the groups dataset.